dify/docs/all.weaviate_integration.how_to_guide.md

```diff
@@ -0,0 +1,187 @@
+      - 50051:50051
+      volumes:
+      - weaviate_data:/var/lib/weaviate
+      restart: on-failure:0
+      environment:
+        QUERY_DEFAULTS_LIMIT: 25
+        AUTHENTICATION_ANONYMOUS_ACCESS_ENABLED: 'true'
+        PERSISTENCE_DATA_PATH: '/var/lib/weaviate'
+        ENABLE_API_BASED_MODULES: 'true'
+        ENABLE_MODULES: ''
+        CLUSTER_HOSTNAME: 'node1'
+      extra_hosts:
+        - "host.docker.internal:host-gateway"
+
+  volumes:
+    weaviate_data:
+  ...
+  ```
+
+- Ollama running on `localhost:11434` with `nomic-embed-text` model
+- Python 3.8+ with required packages installed
+
+## Step 1: Environment Setup
+
+Create a `.env` file with the required configuration:
+
+```bash
+# .env file
+PWD=/path/to/your/project
+OLLAMA_EMBED_URL=http://localhost:11434/api/embed
+OLLAMA_MODEL=nomic-embed-text
+API_KEY=your-secure-api-key-here
+APP_HOST=0.0.0.0
+APP_PORT=2001
+```
+
+## Step 2: Start Required Services
+
+Start Weaviate:
+
+Navigate to weaviate directory where `docker-compose.yml` is present
+
+```bash
+docker compose up -d
+```
+
+Start Ollama and pull the embedding model:
+
+```bash
+ollama serve
+ollama pull nomic-embed-text
+```
+
+## Step 3: Process Your Documents
+
+Use the document processing module to upload your markdown files:
+
+```python
+import logging
+import dify.weaviate_docs as dweadocs
+
+# Configure logging
+dweadocs.configure_logging(level=logging.INFO)
+
+# Process documents
+result = dweadocs.upload_markdown_docs_to_weaviate(
+    docs_dir='./your-docs-directory',
+    collection_name='Documents'
+)
+
+print(f"Successfully processed {result['successful_files']} files")
+```
+
+## Step 4: Start the Retrieval API
+
+Run the FastAPI service that Dify will connect to:
+
+```bash
+python -m dify.weaviate_retrieval
+```
+
+The API will start on `http://localhost:2001` with these endpoints:
+
+- `POST /retrieval` - Main search endpoint for Dify
+- `GET /health` - Health check
+
+## Step 5: Configure Dify External Knowledge
+
+In your Dify application:
+
+1. Go to Knowledge → External Knowledge
+2. Add a new external knowledge source:
+   - API Endpoint: `http://localhost:2001/retrieval`
+   - API Key: The value from your `API_KEY` environment variable
+   - Knowledge ID: `Documents` (the collection name you used)
+
+## Step 6: Test the Integration
+
+Test that everything works:
+Healthcheck:
+```bash
+curl http://localhost:2001/health`
+```
+
+```bash
+# Test the API directly
+curl -X POST http://localhost:2001/retrieval \
+  -H "Authorization: " \
+  -H "Content-Type: application/json" \
+  -d '{
+    "knowledge_id": "Documents",
+    "query": "your search query",
+    "retrieval_setting": {
+      "top_k": 5,
+      "score_threshold": 0.5
+    }
+  }'
+```
+
+## Troubleshooting Quick Fixes
+
+Connection Issues:
+
+- Verify Weaviate:
+  `curl [http://localhost:8080/v1/met](http://localhost:8080/v1/met)a`
+- Verify Ollama:
+  `curl [http://localhost:11434/api/tag](http://localhost:11434/api/tag)s`
```

Comment:
Ugh I think this is a linter bug. Can you pls fix and file an issue showing the problem?